File: src/View/interface.hs
--interface principal do "Who am i?"

-- Menu principal que irá interagir com o usuário
-- O mesmo precisa selecionar opções entre 1 a 4, sendo a opção 1 de iniciar um novo jogo.
menuPrincipal :: IO()
menuPrincipal = do
    putStr ("\n##############################\n")
    putStr ("Bem vindo ao 'Who am i?'!\n\n")
    putStr ("[1] Novo jogo\n")
    putStr ("[2] Como Jogar\n")
    putStr ("[3] Ranking\n")
    putStr ("[4] Sair\n\n")

    opcaoEscolhidaMP <- getLine

    if opcaoEscolhidaMP == 1 then do 
        menuNovoJogo
        else if 
            opcaoEscolhidaMP == 2 then 
                menuComoJogar
        else if 
            opcaoEscolhidaMP == 3 then 
                menuRanking
        else if 
            opcaoEscolhidaMP == 4 then 
                sairPrograma
        else do 
            putStr ("Opção inválida")
            menuPrincipal --para que o programa não termine, e retorne ao mesmo menu.



-- Menu 1, novo jogo
-- Função mais importante, irá popular o savefile de usuários já cadastrados e
-- irá iniciar novo jogo, nas regras de negócio
menuNovoJogo :: IO()
menuNovoJogo : do
    putStr ("[1] Comecar novo jogo\n")
    putStr ("[2] Voltar ao menu principal\n\n")

    opcaoEscolhidaNJ <- getLine

    if opcaoEscolhidaNJ == 1 then do
        putStr("Insira seu nome de usuário\n")
        nome <- getLine -- verificar se, já existe o usuário. Se existir, a pontuacao da partida será somada, do contrário, novo usuário precisa ser criado
        novoJogo -- chamada a funcao do main.hs, iniciando novo jogo e suas respectivas partidas
        else if opcaoEscolhidaNJ == 2 then
            menuPrincipal
        else do
            putStr ("Opção Inválida")
            go to menuNovoJogo --para que o programa não termine, e retorne ao mesmo menu.



-- Menu 2, como jogar
-- Imprime apenas orientações da regra geral de negócio.
menuComoJogar :: IO()
menuComoJogar = do
    putStr ("“Who am i?” é um jogo de turnos, você escolhe um personagem e o Bot irá receber outro personagem aleatoriamente.\n")
    putStr ("A lista de personagens é mesma para ambos.\n" ++ "O objetivo do jogo é adivinhar o personagem do adversário, a cada rodada você pode:\n")
    putStr ("Questionar uma característica ou pode tentar chutar o personagem escolhido pelo Bot.\n" ++ "Para vencer, é necessário adivinhar o personagem primeiro do que o Bot.\n\n\n")
    putStr ("Informações importantes:\n" ++ "Você e o bot podem escolher o mesmo personagem.\nSe você escolher chutar e errar, você perde.\n\n")
    putStr ("[1] Retornar ao Menu Principal\n")

    opcaoEscolhidaCJ <- getLine

    if opcaoEscolhidaCJ == 1 then do 
        menuPrincipal
        else do 
            putStr("Opção inválida")
            menuComoJogar --para que o programa não termine, e retorne ao mesmo menu.



-- Menu 3, ranking de jogadores
-- imprime o save file, ordenado, dos usuários cadastrados. -> USUARIO, Vitorias: X
menuRanking :: IO()
menuRanking = do
    putStr ("Imprime aqui ranking dos top 10 jogadores cadastrados.\n\n") --Falta implementar
    
    putStr ("[1] Retornar ao menu Principal")
    opcaoEscolhidaMR <- getLine

    if opcaoEscolhidaMR == 1 then do 
        menuPrincipal
        else do
            putStr ("Opção inválida")
            menuRanking --para que o programa não termine, e retorne ao mesmo menu.



-- Menu 4 Sair do programa
-- encerra a execucao do programa na opção 1, retorna ao menu principal na posicao 0
sairPrograma :: IO()
sairPrograma = do
    putStr ("Tem certeza que deseja sair?\n")
    putStr ("[1] Sim")
    putStr ("[0] Não")

    opcaoEscolhidaSP <- getLine

    if opcaoEscolhidaSP == 1 then do 
        System.Exit(0)
        else if opcaoEscolhidaSP == 0 then
            menuPrincipal
        else do
            putStr ("Opção inválida")
            sairPrograma --para que o programa não termine, e retorne ao mesmo menu.

File: Prolog/src/Interface.pl
:- module(interface, [
  mainMenu/0,
  mainMenuSwitcher/1,
  newGameMenu/0,
  newGameSwitcher/1,
  tutorial/0
]).

mainMenu:-
  write("\n=========================\n"),
  write("Bem vindo ao `Who Am I?`!\n"),
  write("=========================\n"),
  write("O que voce deseja fazer?\n"),
  write("(1) Iniciar novo jogo\n"),
  write("(2) Listar pontuacoes\n"),
  write("(3) Como Jogar\n"),
  read(Choise), mainMenuSwitcher(Choise).

mainMenuSwitcher(1):- newGameMenu,!.
mainMenuSwitcher(2):- scoreboard,!.
mainMenuSwitcher(3):- tutorial,!.
mainMenuSwitcher(_):- write('Opção inválida!'), mainMenu.


newGameMenu:-
  write("\n=========================\n"),
  write("[1] Player VS BOT \n"),
  write("[2] Player VS Player \n"),
  write("[3] Retornar ao menu principal\n"),
  write("Selecione uma opcao: \n"),
  read(Choise), newGameSwitcher(Choise).

newGameSwitcher(1):- playerXBot,!.
newGameSwitcher(2):- playerXplayer,!.
newGameSwitcher(3):- mainMenu,!.
newGameSwitcher(_):- write('Opção inválida!'), newGameMenu.


tutorial:-
  write("\n==============================================\n"),
  write("Who am i? eh um jogo de turnos, você escolhe um personagem e o Bot ira receber outro personagem aleatoriamente. Onde a lista de personagens eh a mesma para ambos.\n\n"),
  write("O objetivo do jogo eh adivinhar o personagem do adversario, a cada rodada voce pode:\n"),
  write(". Eliminar uma caracteristica\n"),
  write("Para vencer, eh necessario adivinhar o personagem primeiro do que o Bot.\n\n"),
  write("Modos de jogo:\n"),
  write("JogadorxJogador\n"),
  write("JogadorxIA\n\n"),
  write("Informacoes importantes:\n"),
  write("Voce e o bot podem escolher o mesmo personagem.\n"),
  write("Se voce escolher chutar e errar, voce perde a rodada.\n\n"),
  write("[1] Retornar ao menu principal\n"),
  read(Choise), mainMenuSwitcher(Choise).
